django/test/utils.py

```diff
@@ -94,13 +94,23 @@ def instrumented_test_render(self, context):
     return self.nodelist.render(context)
 
 
-def setup_test_environment():
+class _SavedSettings(object):
+    pass
+
+
+def setup_test_environment(debug=None):
     """Perform any global pre-test setup. This involves:
 
         - Installing the instrumented test renderer
         - Set the email backend to the locmem email backend.
         - Setting the active locale to match the LANGUAGE_CODE setting.
     """
+    if debug is None:
```

Comment:
I'm not sure. I guess I could try it and see if it works. Really, there should probably be some sort of dependency injection here to allow testability. In general, I'm not sure we'd want to "bake in" support for calling teardown inside a test (which adding it to a test would do) because there might be global state to be preserved, etc. Does Django have any tests that, for example, call the test runner in a subprocess? That would get around isolation issues, but it would be a much harder test to write.